libcaf_core/caf/actor_system_config.test.cpp

```diff
@@ -28,9 +29,15 @@ using namespace std::literals;
   do {                                                                         \
     using ref_value_type = std::decay_t<decltype(var)>;                        \
     ref_value_type value{__VA_ARGS__};                                         \
-    check_eq(var, value);                                                      \
+    if constexpr (std::is_floating_point_v<decltype(var)>)                     \
+      check_eq(var, test::approx{value});                                      \
+    else                                                                       \
+      check_eq(var, value);                                                    \
```

Comment:
Why not just always compare against `approx{value}`? If it's an integer value, `epsilon` will be 0, so it should turn out the same.

Reply:
If we directly compare against `approx{value}` then we will get compilation error for types where the used operators (>=, <=, + and - ) are not supported. Due to this, this floating check has been added. I have moved this floating point check inside `friend bool operator==(T lhs, approx<T> rhs)` to fix address the feedback.